src/linpde_gp/domains/_box.py

```diff
@@ -73,24 +78,37 @@ def __contains__(self, item: ArrayLike) -> bool:
     def __eq__(self, other) -> bool:
         return isinstance(other, Box) and np.all(self.bounds == other.bounds)
 
-    def uniform_grid(self, shape: ShapeLike, inset: ArrayLike = 0.0) -> np.ndarray:
-        shape = pn.utils.as_shape(shape, ndim=len(self._interior_idcs))
-        insets = np.broadcast_to(inset, len(self._interior_idcs))
+    def uniform_grid(
+        self, shape: ShapeLike, inset: ArrayLike = 0.0
+    ) -> "TensorProductGrid":
+        from linpde_gp.randprocs.covfuncs import (  # pylint: disable=import-outside-toplevel
+            TensorProductGrid,
+        )
+
+        total_dim = len(self.bounds)
+
+        interior_shape = pn.utils.as_shape(shape, ndim=len(self._interior_idcs))
+        interior_inset = np.broadcast_to(inset, len(self._interior_idcs))
+        total_shape = np.ones((total_dim,), dtype=int)
+        total_insets = np.zeros((total_dim,), dtype=float)
 
-        noncollapsed_grids = np.meshgrid(
+        for i, interior_idx in enumerate(self._interior_idcs):
+            total_shape[interior_idx] = interior_shape[i]
+            total_insets[interior_idx] = interior_inset[i]
```

Comment:
Wouldn't
```suggestion
        total_shape[self._interior_idcs] = interior_shape
        total_insets[self._interior_idcs] = interior_inset
```
work too?